schema/schemaspec/spec.go

```diff
@@ -43,6 +43,12 @@ type (
 	Ref struct {
 		V string
 	}
+
+	// Unmarshaler is the interface that wraps the Unmarshal function.
+	Unmarshaler interface {
+		// Unmarshal decodes an Atlas DDL document into target.
+		Unmarshal(target interface{}) error
```

Comment:
https://pkg.go.dev/encoding/json#Unmarshaler
https://pkg.go.dev/encoding/xml#Unmarshaler
```suggestion
		UnmarshalSpec(target interface{}) error
```

Can you elaborate on the input type? Why it's not `[]byte`?